apps/api/src/app/change/usecases/change-enabled-notification-template/change-enabled-notification-template.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { TypeChangeEnabledCommand } from '../type-change-enabled.command';
+import { NotificationTemplateEntity, NotificationTemplateRepository, MessageTemplateRepository } from '@novu/dal';
+
+@Injectable()
+export class ChangeEnabledNotificationTemplate {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private messageTemplateRepository: MessageTemplateRepository
+  ) {}
+
+  async execute(command: TypeChangeEnabledCommand) {
+    const item = await this.notificationTemplateRepository.findOne({
+      _environmentId: command.environmentId,
+      _parentId: command.item._id,
+    });
+
+    const newItem = command.item as NotificationTemplateEntity;
+
+    const messages = await this.messageTemplateRepository.find({
+      _environmentId: command.environmentId,
+      _parentId: {
+        $in: newItem.steps ? newItem.steps.map((step) => step._id) : [],
+      },
+    });
+
+    const missingMessages = [];
+    const steps = newItem.steps
+      ? newItem.steps
+          .map((step) => {
+            const oldMessage = messages.reduce((prev, message) => {
+              return message._parentId === step._id ? message : prev;
+            }, undefined);
+
+            if (!oldMessage) {
+              missingMessages.push(step._id);
+
+              return undefined;
+            }
+            step._id = oldMessage._id;
+
+            return step;
+          })
+          .filter((step) => step !== undefined)
+      : [];
```

Comment:
This is kind of a big ternary statement :) Maybe we can extract the first part to a variable? It makes it a bit hard to understand what is going on here

Reply:
I am not sure I understand your suggestion...

Reply:
Oh sorry, let me attach an example.

```
const steps = newItem.steps?
          .map((step) => {
            const oldMessage = messages.reduce((prev, message) => {
              return message._parentId === step._id ? message : prev;
            }, undefined);
davidsoderberg marked this conversation as resolved.

            if (!oldMessage) {
              missingMessages.push(step._id);

              return undefined;
            }
            step._id = oldMessage._id;

            return step;
          })
          .filter((step) => step !== undefined) || [];
```

Or something like this ^
Or maybe:
```
function mapNewStepItem(step: StepType) {
    const oldMessage = messages.reduce((prev, message) => {
      return message._parentId === step._id ? message : prev;
    }, undefined);
davidsoderberg marked this conversation as resolved.

    if (!oldMessage) {
      missingMessages.push(step._id);

      return undefined;
    }
    step._id = oldMessage._id;

    return step;
}

const steps = newItem.steps?
          .map(mapNewStepItem)
          .filter((step) => step !== undefined) || [];
```